napari/_qt/qt_main_window.py

```diff
@@ -491,6 +491,32 @@ def changeEvent(self, event):
 
         super().changeEvent(event)
 
+    def keyPressEvent(self, event):
+        """Called whenever a key is pressed.
+
+        Parameters
+        ----------
+        event : qtpy.QtCore.QEvent
+            Event from the Qt context.
+        """
+        self._qt_viewer.canvas._backend._keyEvent(
+            self._qt_viewer.canvas.events.key_press, event
```

Comment:
```suggestion
        self._qt_viewer.canvas._scene_canvas._backend._keyEvent(
            self._qt_viewer.canvas._scene_canvas.events.key_press, event
```

Reply:
why?

Reply:
Because of the traceback I get?
https://github.com/napari/napari/pull/6419#issuecomment-1800893574
I think your code is for 0.4.19 not main